samples/app/src/index.js

```diff
@@ -8,31 +10,21 @@ import './index.css';
 // it's up to you to detect the locale of your user
 // (by url, cookie, header X-Accept-Language, etc)
 const matches = document.location.search.match(/lang=(.*)/);
-const locale = (Array.isArray(matches) && matches.pop()) || 'en';
+let locale = (Array.isArray(matches) && matches.pop()) || 'en';
+
+const availableLocales = ['fr', 'en'];
+if (!availableLocales.includes(locale)) {
+  locale = 'en';
+}
```

Comment:
Sounds `navigator.languages` has not enough support, don't you think?

There's a typo, `negociateLocale` is not recursive:

```
const userBrowserLocales = window.navigator.languages || [(window.navigator.language || window.navigator.userLanguage)];
const locale = negociateLocale([
      queryParamLocale,
      ...userBrowserLocales,
    ], availableLocales, defaultLocale);
```